Assets/HoloToolkit-Examples/SpectatorView/Scripts/ColorChanger.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.using UnityEngine;
+
+using System;
+using UnityEngine;
+using UnityEngine.Networking;
+
+namespace HoloToolkit.Examples.SpectatorView
+{
+    /// <summary>
+    /// Network component that randomly changes the color of its Renderer every given time
+    /// </summary>
+    public class ColorChanger : NetworkBehaviour
+    {
+        /// <summary>
+        /// Current color iteration of the object
+        /// </summary>
+        [SyncVar] private Color color;
+
+        /// <summary>
+        /// Renderer to which the color is applied to
+        /// </summary>
+        private Renderer objectRenderer;
+
+        /// <summary>
+        /// Counts the time between color changes
+        /// </summary>
+        private float timer;
+
+        private void Start()
+        {
+            objectRenderer = GetComponent<Renderer>();
+            UnityEngine.Random.InitState(DateTime.Now.Millisecond);
+            ChangeColor();
+        }
+
+        private void Update()
+        {
+            if (isServer)
+            {
+                if (timer > 3.0f)
+                {
+                    ChangeColor();
+                    timer = 0f;
+                }
+
+                timer += Time.deltaTime;
+            }
+
+            objectRenderer.material.color = color;
+        }
+
+        /// <summary>
+        /// Changes the renderer to a new random color
+        /// </summary>
+        private void ChangeColor()
+        {
+            color = UnityEngine.Random.ColorHSV();
+            objectRenderer.material.color = color;
```

Comment:
Or cleanup the `material` in `OnDestroy`.
According to Unity, it's [not recommended](https://docs.unity3d.com/ScriptReference/Renderer-sharedMaterial.html) to modify the materials returned by sharedMaterial, since it changes all instances of that material and changes the actual material file in the project too. They do [have a note](https://docs.unity3d.com/ScriptReference/Renderer-material.html) about cleaning up the material:
>This function automatically instantiates the materials and makes them unique to this renderer. It is your responsibility to destroy the materials when the game object is being destroyed.